docs/reference/search/search-your-data/search-template.asciidoc

```diff
@@ -630,3 +631,453 @@ POST _render/template
   }
 }
 ----
+[[search-template-with-mustache-examples]]
+=== Search template examples with Mustache
+Mustache template help with running your search template with different variables. You can set your value to variable through `params` in template.
+
+This document illustrates more examples of mustache tags with search templates.
+
+[discrete]
+[[search-template-mustache-variable]]
+=== Mustache Variable
+Mustache tags are typically enclosed in double curly brackets, A mustache variable : `{{my-variable}}` is a type of mustache tag which checks for a matching `my-variable` key in your search template `params` and replaces template with the value set for `my-variable`
+
+For example, following search template replaces `{{query_string}}` in `message` with `My query string` value passed in `params`
+
+[source,console]
+
+----
+
+PUT _scripts/my-search-template
+{
+  "script": {
+    "lang": "mustache",
+    "source": {
+      "query": {
+        "match": {
+          "message": "{{query_string}}"
+        }
+      },
+      "from": "{{from}}",
+      "size": "{{size}}"
+    },
+    "params": {
+      "query_string": "My query string"
+    }
+  }
+}
+
+----
+
+Testing `my-search-template` with different `params`
+
+[source,console]
+
+----
+
+
+POST _render/template
+{
+  "id": "my-search-template",
+  "params": {
+    "query_string": "hello world",
+    "from": 20,
+    "size": 10
+  }
+}
+
+
+----
+
+When rendered, the template outputs as:
+[source,console-result]
+
+----
+
+{
+  "template_output": {
+    "query": {
+      "match": {
+        "message": "hello world"
+      }
+    },
+    "from": "20",
+    "size": "10"
+  }
+}
+
+----
+
+If your search template doesn't pass a value to your `query_string` the message would be replaced with a empty string.
+
+[source,console]
+
+----
+
+POST _render/template
+{
+  "id": "my-search-template",
+  "params": {
+    "from": 20,
+    "size": 10
+  }
+}
+
+----
+
+When rendered, template outputs as:
+[source,console-result]
+
+----
+
+{
+  "template_output": {
+    "query": {
+      "match": {
+        "message": ""
+      }
+    },
+    "from": "20",
+    "size": "10"
+  }
+}
+
+----
+
+[discrete]
+[[search-template-sections]]
+==== Sections
+
+Sections are also a type of Mustache tags and are useful when you have to loop through nested keys. A sections begins with `{{#my-section-variable}}` and ends with `{{\my-section-variable}}`
```

Comment:
Good catch. Sections are similar to conditionals and we could loop through the nested keys. But, the example provided for `sections` doesn't show looping. But I do show the example in LIST. 

Anyway, I guess I will rephrase the sentence a bit and  may be alter example to avoid confusions.